critiquebrainz/db/oauth_client.py

```diff
@@ -0,0 +1,137 @@
+from critiquebrainz import db
+from critiquebrainz.db import exceptions as db_exceptions
+from critiquebrainz.utils import generate_string
+import sqlalchemy
+
+
+def create(*, user_id, name, desc, website, redirect_uri):
+    """Create new OAuth client and generates secret key for it.
+
+    Args:
+        user_id (uuid): ID of the user who manages the client.
+        name (str): Name of the client.
+        desc (str): Client description.
+        website (str): Client web site.
+        redirect_uri: URI where responses will be sent.
+    Returns:
+        Dictionary of the following structure
+        {
+            "client_id": str,
+            "client_secret": str,
+            "user_id": uuid,
+            "name": str,
+            "desc": desc,
+            "website": website,
+            "redirect_uri": redirect_uri,
+        }
+    """
+    client_id = generate_string(20)
+    client_secret = generate_string(40)
+    client = {
+        "client_id": client_id,
+        "client_secret": client_secret,
+        "redirect_uri": redirect_uri,
+        "user_id": user_id,
+        "name": name,
+        "website": website,
+        "desc": desc,
+    }
+    with db.engine.connect() as connection:
+        connection.execute(sqlalchemy.text("""
+            INSERT INTO oauth_client VALUES(:client_id, :client_secret, :redirect_uri,
+                :user_id, :name, :desc, :website)
+        """), client)
+
+    return client
```

Comment:
Is the return value being used anywhere outside of tests? If not, you can remove it and move dictionary directly into the `connection.execute` call.